### DISCUSSION TITLE
Why is Focal Loss using a sigmoid activation even in the multiclass case?

### DISCUSSION THREAD

**Discussion by jmv115** [qubvel/segmentation_models.pytorch#1232] [Q&A]
I found this stale issue that touches upon this same topic. https://github.com/qubvel-org/segmentation_models.pytorch/issues/671 

Currently the `losses.FocalLoss` interface doesn't allow to pass probabilities (unlike other Losses that have both a probs and logits implementation) and expects logits, which then are activated by a sigmoid in the `losses._functional.focal_loss_with_logits` function. Since sigmoid can only work in a binary case, the loss is calculated per class in binary fashion and then summed.

Interestingly, there's also a softmax version in `losses._functional.softmax_focal_loss_with_logits` and it's more similar to how the FocalLoss works in segmentation_models for tensorflow/keras. In this case, the loss wouldn't need to be calculated per class in the wrapper.

Is there any reason on why Focal Loss is calculated with this function and using sigmoid? Should the results be equivalent to softmax somehow? I would think softmax would be a more appropiate solution for the multiclass case but I'm am all ears in case this is a nicer approach.
